pc/storage/kdtree/kdtree_test.go

```diff
@@ -198,28 +279,351 @@ func TestKDtree(t *testing.T) {
 			})
 		}
 	})
-}
 
-func TestKDtree_randomCloud(t *testing.T) {
-	const (
-		nPoints = 100
-		width   = 10.0
-	)
+	t.Run("FindMinimum", func(t *testing.T) {
+		testCases := []struct {
+			dim    int
+			nodeID int
+		}{
+			{dim: 0, nodeID: 4},
+			{dim: 1, nodeID: 3},
+			{dim: 2, nodeID: 3},
+			{dim: 3, nodeID: -1},
+		}
 
-	pp := generateRandomCloud(t, nPoints, width)
-	it, err := pp.Vec3Iterator()
-	if err != nil {
-		t.Fatal(err)
+		for _, tt := range testCases {
+			tt := tt
+			t.Run(fmt.Sprintf("dim: %d", tt.dim), func(t *testing.T) {
+				nID, err := kdt.findMinimumImpl(kdt.root, tt.dim, 0)
+				if nID != tt.nodeID {
+					t.Errorf("Expected %v, got: %v", tt.nodeID, nID)
+				}
+				if tt.dim > 2 && err == nil {
+					t.Errorf("Expected an error when dim>2")
+				}
+			})
+		}
+	})
+
+	t.Run("deleteNodeImpl", func(t *testing.T) {
+		it := createTestPointCloud(t)
+		var kdt *KDTree
+		testCases := map[string][]struct {
+			pID          int
+			expectedTree *KDTree
+		}{
+			"leaf then node with right sub tree": {
```

Comment:
Go's test name can't contain whitespace and automatically replaced by `_`.
Personally, I prefer CamelCase test name.